WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java

```diff
@@ -687,33 +690,14 @@ private void moderateComment(
         // Fire the appropriate listener if we have one
         if (note != null && mOnNoteCommentActionListener != null) {
             mOnNoteCommentActionListener.onModerateCommentForNote(note, newStatus);
-            dispatchModerationAction(site, comment, newStatus);
+            mViewModel.dispatchModerationAction(site, comment, newStatus);
         } else if (mOnCommentActionListener != null) {
             mOnCommentActionListener.onModerateComment(comment, newStatus);
             // Sad, but onModerateComment does the moderation itself (due to the undo bar), this should be refactored,
```

Comment:
According to this comment, we cannot dispatchModerationAction when the comment is from comment list. So I keep the current behavior, every time we moderate the comment, it will finish the screen (when it is from comment list).

Reply:
Sounds good 👍